src/components/ProgramCards.tsx

```diff
@@ -19,53 +19,54 @@ const ChoseColor = () => {
 };
 
 const ProgramCards = ({
-	programs,
 	onOpenRedeployProgram,
 	isRedeployLoading,
-	setSelectedProgram,
-}: ProgramCardsProps): JSX.Element => (
-	<>
-		{programs.map((program: IPCProgram) => (
-			<ProgramCard key={program.createdAt} program={program}>
-				<>
-					<Box w="25%">
-						<HStack>
-							<FcRules display="flex" size="40"></FcRules>
-							<Text textColor={ChoseColor()}>{program.name}</Text>
-						</HStack>
-					</Box>
-					<Box w="45%">
-						<Button
-							as="a"
-							href={`https://aleph.sh/vm/${program.hash}`}
-							target="_blank"
-							variant="inline"
-							size="sm"
-							w="45%"
-							p="0px"
-							id="ipc-computing-forwardUrl-button"
-							marginRight="10"
-						>
-							Go to site
-						</Button>
-						<RedeployButton
-							onClick={() => {
-								setSelectedProgram(program);
-								onOpenRedeployProgram();
-							}}
-							isLoading={isRedeployLoading}
-						/>
-					</Box>
-					<Text textColor={ChoseColor()}>
-						{`${new Date(program.createdAt).toString().substring(4, 15).slice(0, 3)} /${new Date(program.createdAt)
-							.toString()
-							.substring(4, 15)
-							.slice(3, 6)} /${new Date(program.createdAt).toString().substring(4, 15).slice(6)}`}
-					</Text>
-				</>
-			</ProgramCard>
-		))}
-	</>
-);
+}: ProgramCardsProps): JSX.Element => {
+	const { programs } = useDriveContext();
+
+	return (
+		<>
+			{programs.map((program: IPCProgram) => (
+				<ProgramCard key={program.createdAt} program={program}>
+					<>
+						<Box w="25%">
+							<HStack>
+								<FcRules display="flex" size="40"></FcRules>
+								<Text textColor={ChoseColor()}>{program.name}</Text>
+							</HStack>
+						</Box>
+						<Box w="45%">
+							<Button
+								as="a"
+								href={`https://aleph.sh/vm/${program.hash}`}
+								target="_blank"
+								variant="inline"
+								size="sm"
+								w="45%"
+								p="0px"
+								id="ipc-computing-forwardUrl-button"
+								marginRight="10"
+							>
+								Go to site
+							</Button>
+							<RedeployButton
+								onClick={() => {
+									onOpenRedeployProgram();
+								}}
+								isLoading={isRedeployLoading}
+							/>
+						</Box>
+						<Text textColor={ChoseColor()}>
+							{`${new Date(program.createdAt).toString().substring(4, 15).slice(0, 3)} /${new Date(program.createdAt)
+								.toString()
+								.substring(4, 15)
+								.slice(3, 6)} /${new Date(program.createdAt).toString().substring(4, 15).slice(6)}`}
```

Comment:
It would be better to create a function in `utils` for this with comments, it's quite unreadable here :frowning_face: